pydantic_ai_slim/pydantic_ai/_output.py

```diff
@@ -399,6 +504,11 @@ def __init__(
             allows_deferred_tools=allows_deferred_tools,
             allows_image=allows_image,
         )
+        self.json_schema = OutputSchema[OutputDataT].build_json_schema(
```

Comment:
I don't think we need to proactively build it -- I'd rather have it be a `cached_property`

Reply:
Also, could it be implemented on `OutputSchema`, and take into account all the `self.allows_X` fields to build itself?

Reply:
I can implement on `OutputSchema` maybe, but I think it would be ugly because of the way `processor` is only built for some classes: https://github.com/pydantic/pydantic-ai/blob/93d5f77a9a3078a05a93d5b59e5bdc68a1f27ee2/pydantic_ai_slim/pydantic_ai/_output.py#L267-L272. Perhaps there is a clean way to do it that I don't see?

Reply:
Okay the above statement is wrong, I misremembered. The challenge I am having is that `allows_text` is always true for the `AutoOutputSchema`, so the union always has type `str` in it. Is it supposed to be that way?

Reply:
I figured out how to get it to work without using `allows_text`.